doc/modules/preprocessing.rst

```diff
@@ -551,6 +551,42 @@ values than observed values.
 :class:`Imputer` can be used in a Pipeline as a way to build a composite
 estimator that supports imputation. See :ref:`sphx_glr_auto_examples_plot_missing_values.py`.
 
+The :class:`KNNImputer` class provides imputation for completing missing
+values using the k-Nearest Neighbors approach. Each sample's missing values
+are imputed from up to n_neighbors nearest neighbors found in the training set.
+Each missing feature is then imputed as the average, either weighted or
+unweighted, of the neighbors who have a value for it.
+When any of the neighbors themselves have the feature value missing then
```

Comment:
It is an average, but in case some of the neighbors are missing, it is the re-scaled average of the remaining neighbors. Re-scaling involves increasing the weights of the remaining neighbors by the factor noted in the formula.